docs/modeling/models.rst

```diff
@@ -424,240 +424,135 @@ the number of models, with one input per model::
-   plt.show()
-
-- When passed a 2-D array, the same polynomial will map each row of the array
-  to one model in the set, one for one::
-
-      >>> x = np.arange(30).reshape(5, 6)
-      >>> y = p1(x)
-      >>> y  # doctest: +FLOAT_CMP
-      array([[  0.,   0.,   0.,   0.,   0.,   0.],
-             [  6.,   7.,   8.,   9.,  10.,  11.],
-             [ 24.,  26.,  28.,  30.,  32.,  34.],
-             [ 54.,  57.,  60.,  63.,  66.,  69.],
-             [ 96., 100., 104., 108., 112., 116.]])
-      >>> y.shape
-      (5, 6)
+There are two ways to get around this. ``model_set_axis`` can be passed in
+when the model is evaluated::
+
+    >>> g(x, model_set_axis=1)
+    array([[1.        , 1.76499381],
+           [1.        , 1.76499381],
+           [1.        , 1.76499381]])
+
+Or when the model is initialized::
+
+    >>> g = models.Gaussian1D(amplitude=[[1, 2]], mean=[[0, 0]],
+    ...                       stddev=[[0.1, 0.2]], n_models=2,
+    ...                       model_set_axis=1)
+    >>> g(x)
+    array([[1.        , 1.76499381],
+           [1.        , 1.76499381],
+           [1.        , 1.76499381]])
+
+Note that in the latter case the shape fo the individual parameters has changed to 2D.
+
+The value of ``model_set_axis`` is either an integer number, representing the axis along which
+the different parameter sets and inputs are defined, or a boolean of value ``False``,
+in which case it indicates all model sets should use the same inputs on evaluation.
+For example the above model has a value of 1 for ``model_set_axis``.
+If ``model_set_axis=False`` is passed the 2 models will be evaluated on the same input::
+
+    >>> g.model_set_axis
+    1
+    >>> result = g(x, model_set_axis=False)
+    >>> result
+    array([[[1.        , 0.60653066],
+            [2.        , 1.76499381]],
+    <BLANKLINE>
+           [[1.        , 0.60653066],
+            [2.        , 1.76499381]],
+    <BLANKLINE>
+           [[1.        , 0.60653066],
+            [2.        , 1.76499381]]])
+    >>> result[: , 0]
+    array([[1.        , 0.60653066],
+           [1.        , 0.60653066],
+           [1.        , 0.60653066]])
+    >>> result[: , 1]
+    array([[2.        , 1.76499381],
+           [2.        , 1.76499381],
+           [2.        , 1.76499381]])
+
+Currently model sets are most useful for fitting a set of **linear** models
+(:ref:`example <example-fitting-model-sets>`)
+allowing a large number of models of the same type to be fitted simultaneously
+(and independently from each other) to some large set of inputs, for example,
+fitting a polynomial to the time response of each pixel in a data cube.
+This can greatly speed up the fitting process. The speed-up is due to solving
+the set of equations to find the exact solution. Non-linear models, which require
+an iterative algorithm, cannot be currently fit using model sets. Model sets of non-linear
+models can only be evaluated.
+
+
+.. _modeling-asdf:
+
+Model Serialization
```

Comment:
"Serialization" is a very accurate term, but perhaps "Saving Models" or "Writing Models to Disk" might be a more accessible way to convey this to people who don't use the term "serialize" every day, but still want to write their models out (and read them back in).